File: leetcode-arry/80. 删除有序数组中的重复项 II/python/remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        count = 0
        for right in range(len(nums)):
            if count < 2 or nums[count - 2] != nums[right]:
                nums[count] = nums[right]
                count += 1
        return count

File: leetcode-arry/80. 删除有序数组中的重复项 II/java/Remove Duplicates from Sorted Array II.java
class Solution{
    public int removeDuplicates(int[] nums) {
        int i = 0;
        for(int n : nums) {
            if(i < 2 || n > nums[i - 2]) {
                nums[i++] = n;
            }
        }
        return i;
    }

    public int doublePointer(int[] nums) {
        int left = 0;
        int count = 0;
        for(int right = 1;right < nums.length;right++) {
            if(nums[left] != nums[right]) {
                nums[++left] = nums[right];
                count = 0;
            } else if(count >= 1) {
                continue;
            } else {
                nums[++left] = nums[right];
                count++;
            }
        }
        return left + 1;
    }
}